tests/test_infographic_api.py

```diff
@@ -0,0 +1,304 @@
+import pytest
+import six
+from six.moves import http_client
+from anyway import app as flask_app
+from jsonschema import validate
+
+
+@pytest.fixture
+def app():
+    return flask_app.test_client()
+
+
+if six.PY2:
+    def _text_data(rv): return rv.data
+else:
+    def _text_data(rv): return rv.data.decode("utf-8")
+
+
+class TestInfographicApi:
+    @pytest.fixture
+    def infographic_from_existing_new_flash(self, app):
+        EXITING_NEWS_FLASH_ID = 15305
+
+        rv = app.get(
+            f'/api/infographics-data?news_flash_id={EXITING_NEWS_FLASH_ID}')
+
+        return rv.get_json()
+
+    @pytest.fixture
+    def widgets(self, infographic_from_existing_new_flash):
+        return infographic_from_existing_new_flash['widgets']
+
+    def test_no_news_flash_id(self, app):
+        '''Should success and be empty when no flash id is sent'''
+        rv = app.get('/api/infographics-data')
+        assert rv.status_code == http_client.OK
+        assert not _text_data(rv)
+
+    def test_bad_news_flash_id(self, app):
+        '''Should success and be empty when bad news flash id'''
+        rv = app.get('/api/infographics-data?news_flash_id=-1')
+        assert rv.status_code == http_client.OK
+        assert not _text_data(rv)
+
+    def test_location_info(self, infographic_from_existing_new_flash):
+        assert infographic_from_existing_new_flash['meta']['location_info'] == {
+            "resolution": "כביש בינעירוני",
+            "road1": 90.0,
+            "road_segment_name": "כניסה למצפה שלם - צומת שדי תרומות"
+        }
+
+    def test_infographic_with_existing_news_flash(self, widgets):
```

Comment:
IMO the widget should be a constant, 
All those tests are going to break the moment the value return by the website will be different,
while the functionality under tests will stay the same.

Reply:
We talked about this